Short-Answer/Algorithms_Answers.md

```diff
@@ -2,14 +2,14 @@
 
 ## Exercise I
 
-a)
+a) O(n^c) Polynomial time. The inner functions seems to be efficient code, but the outer while loop will greatly increase the number of operations.
```

Comment:
Nice job on all of these!!

With this one I'm curious how you arrived at `O(n^c)`. I haven't seen too many Polynomial run-time algos.

`a` in this example is acting as `i` would in a for-loop:

```python
a)  a = 0
    while (a < n * n * n):
      a = a + n * n
```

If you treat it as such, then you can say the while-loop runs at `O(n^3)`.

However, `a = a + n * n` increases `a` at an exponential rate, therefore if you divide `n^3` by `n^2` you're left with just `n`.

The run-time complexity would therefore be `O(n)`.